File: common/src/main/java/dev/number6/comprehend/results/PresentableSentimentResults.java
package dev.number6.comprehend.results;

import com.amazonaws.services.comprehend.model.DetectSentimentResult;

import java.time.LocalDate;
import java.util.Collection;
import java.util.Map;

public class PresentableSentimentResults extends ComprehensionResults<Collection<DetectSentimentResult>> {
    private final SentimentResultsToMessageSentimentScore sentimentResultsToMessageSentimentScore;
    private final SentimentResultsToMessageSentimentTotals sentimentResultsToMessageSentimentTotals;

    public PresentableSentimentResults(LocalDate comprehensionDate,
                                       Collection<DetectSentimentResult> sentimentResults,
                                       String channelName) {
        this(comprehensionDate, sentimentResults, channelName,
                new SentimentResultsToMessageSentimentScore(),
                new SentimentResultsToMessageSentimentTotals());
    }

    public PresentableSentimentResults(LocalDate comprehensionDate,
                                       Collection<DetectSentimentResult> sentimentResults,
                                       String channelName,
                                       SentimentResultsToMessageSentimentScore sentimentResultsToMessageSentimentScore,
                                       SentimentResultsToMessageSentimentTotals sentimentResultsToMessageSentimentTotals) {
        super(comprehensionDate, sentimentResults, channelName);
        this.sentimentResultsToMessageSentimentScore = sentimentResultsToMessageSentimentScore;
        this.sentimentResultsToMessageSentimentTotals = sentimentResultsToMessageSentimentTotals;
    }

    public Map<String, Integer> getSentimentTotals() {
        return sentimentResultsToMessageSentimentTotals.apply(getResults());
    }

    public Map<String, Float> getSentimentScoreTotals() {
        return sentimentResultsToMessageSentimentScore.apply(getResults());
    }

    @Override
    public String toString() {
        return "SentimentResults{" +
                "comprehensionDate=" + getComprehensionDate() +
                ", results=" + getResults() +
                ", channelName='" + getChannelName() + '\'' +
                '}';
    }
}

File: common/src/main/java/dev/number6/comprehend/results/PresentableSentimentResults.kt
package dev.number6.comprehend.results

import com.amazonaws.services.comprehend.model.DetectSentimentResult
import java.time.LocalDate

class PresentableSentimentResults @JvmOverloads constructor(comprehensionDate: LocalDate,
                                                            sentimentResults: Collection<DetectSentimentResult>,
                                                            channelName: String,
                                                            private val sentimentResultsToMessageSentimentScore: SentimentResultsToMessageSentimentScore =
                                                                    SentimentResultsToMessageSentimentScore(),
                                                            private val sentimentResultsToMessageSentimentTotals: SentimentResultsToMessageSentimentTotals =
                                                                    SentimentResultsToMessageSentimentTotals()) :
        ComprehensionResults<Collection<DetectSentimentResult>>(comprehensionDate, sentimentResults, channelName) {
    val sentimentTotals: Map<String, Int>
        get() = sentimentResultsToMessageSentimentTotals.apply(results)

    val sentimentScoreTotals: Map<String, Float>
        get() = sentimentResultsToMessageSentimentScore.apply(results)

    override fun toString(): String {
        return "SentimentResults{" +
                "comprehensionDate=" + comprehensionDate +
                ", results=" + results +
                ", channelName='" + channelName + '\'' +
                '}'
    }

}